mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.java

```diff
@@ -1069,6 +1069,8 @@ public void bind(Widget widget) {
                     mConnection.getUsername(), mConnection.getPassword()));
             mWebView.getSettings().setDomStorageEnabled(true);
             mWebView.getSettings().setJavaScriptEnabled(true);
+            mWebView.getSettings().setSupportMultipleWindows(true);
+            mWebView.setBackgroundColor(Color.TRANSPARENT);
```

Comment:
This breaks websites that rely on having a white background and don't set one in CSS, e.g. default Apache HTTP error sites.
Maybe there's a way to set the webview background to transparent only if the website background is set to transparent.

Reply:
Errr... After all it's a widget on a control panel, it's not a generic browser. Aren't we supposed to put only specially designed content there?
In other words: is there any real-world case which would break?
 I'll have a look if there's a cheap way to dig into loaded DOM.

Reply:
I've looked at this from different perspectives.
1. Indeed. If a web site doesn't specify a background, it will be taken from what we set for the WebView.
2. Plain text is still readable using "dark" theme. So the user will see 404.
3. There's no reliable way to inspect the whole DOM inside the WebView. You could only get HTML as a string, but this then means you would need to parse the whole thing, perhaps loading external CSS... Ending up in reimplementing your own browser. Using some magic tag/comment would be a hack and again you can think of a case where this woudn't work (e. g. the widget contents is hosted completely elsewhere).
4. If you still think it's an issue, a cheap way to solve this would be to add a setting.
5. Perhaps the most user-friendly and flexible way would be to allow to specify backround RGBA value in the sitemap for a particular widget. But i guess this would be hard to implement since it requires changes to OpenHAB core too.
 So what's your opinion?
 BTW, HABPanel does use transparent background and apparently nobody suffers.

Reply:
> But i guess this would be hard to implement since it requires changes to OpenHAB core too.

If something _has to be_ changed in openHAB core, you can open an issue there. Something like `transparentBackground=true` to set the background transparent. Defaults to false. Let's discuss it there as it also affects Basic UI.

>  BTW, HABPanel does use transparent background and apparently nobody suffers.

I guess no one suffers, because HABPanel has it's own background color.

Reply:
Well, BasicUI has it too.

Reply:
https://github.com/openhab/openhab-core/issues/809